### ISSUE TITLE
SC2001 should recommend non-global equivalent for non-global sed expressions

### ISSUE THREAD

**Issue by tellowkrinkle** [koalaman/shellcheck#2164]
#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
file=a/a/a
echo "$file" | sed -e 's/a/b/'
```

#### Here's what shellcheck currently says:

`SC2001: See if you can use ${variable//search/replace} instead.`

#### Here's what I wanted or expected to see:

`SC2001: See if you can use ${variable/search/replace} instead.`

Replacing `echo "$file" | sed -e 's/a/b/'` with `echo "${file//a/b}` changes the output from `b/a/a` to `b/b/b`

**Comment by systemmonkey42** [comment]
Does shellcheck pay attention to what is in the SED expression?

Unless I'm mistaken (and I often am) ${variable/search/replace} is not a REGEX replacement, as such I would have thought is was rare that ${variable/search/replace} could be equivalent a `sed` expression.

More importantly, If the "search" part of the SED replacement contains ANY unescaped regex characters, the warning should be disabled.
